qi-lib/flow/core/normalize.rkt

```diff
@@ -1,78 +1,85 @@
 #lang racket/base
 
-(provide normalize-rewrite)
-
-(require syntax/parse
-         (for-template racket/base))
+(require (for-syntax racket/base
+                     syntax/parse
+                     "pass.rkt"
+                     "private/form-property.rkt")
+         "passes.rkt")
 
 ;; 0. "Qi-normal form"
-(define (normalize-rewrite stx)
-  (syntax-parse stx
-    #:datum-literals (#%host-expression
-                      #%blanket-template
-                      #%fine-template
-                      esc
-                      gen
-                      thread
-                      pass
-                      if
-                      amp
-                      relay
-                      tee
-                      sep
-                      collect
-                      __)
+(begin-for-syntax
```

Comment:
Why remove `normalize-rewrite`? Feels useful to have the single rewrite as a distinct function and then have the pass call it to a fixed point. So I think, below, it would be `(fix normalize-rewrite)`